library/vyos_interfaces.py

```diff
@@ -0,0 +1,396 @@
+###
+##############################################
+##############################################
+##############################################
+
+"""
+The module file for vyos_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'network'
+}
+
+DOCUMENTATION = """
+---
+module: vyos_interfaces
+version_added: 2.9
+short_description: Manages interface attributes of VyOS network devices.
+description: This module manages the interface attributes on VyOS network devices.
+author: Nilashish Chakraborty (@nilashishc)
+options:
+  config:
+    description: The provided configuration
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+        - Full name of the interface, e.g. eth0, eth1.
+        type: str
+        required: True
+      description:
+        description:
+          - Interface description.
+        type: str
+      duplex:
+        description:
+          - Interface duplex mode. Applicable for Ethernet interfaces only.        
+        choices: ['full', 'half', 'auto']
+      enable:
+        default: True
+        description:
+          - Administrative state of the interface. 
+          - Set the value to C(true) to administratively enable the interface or C(false) to disable it.
+        type: bool
+      mtu:
+        description:
+          - MTU for a specific interface. Must be an even number between 68 to 9000.
+          - Applicable for Ethernet interfaces only.
+        type: str
+      speed:
+        description:
+          - Interface link speed. Applicable for Ethernet interfaces only.
+        type: str
+        choices: ['auto', '10', '100', '1000', '2500', '10000']
+      vifs:
+        description: 
+          - Virtual sub-interfaces related configuration.
+          - 802.1Q VLAN interfaces are represented as virtual sub-interfaces in VyOS.
+        elements: dict
+        type: list
+        suboptions:
+          vlan_id:
+            description:
+              - Identifier for the virtual sub-interface.
+            type: str
+          description: 
+            description: 
+              - Virtual sub-interface description.
+            type: str
+          enable: 
+            description:
+              - Administrative state of the virtual sub-interface.
+              - Set the value to C(true) to administratively enable the interface or C(false) to disable it.
+            type: bool
+            default: True
+          mtu:
+            description: 
+              - MTU for the virtual sub-interface. Must be an even number between 68 to 9000.
+            type: str
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    default: merged
+    
+"""
+EXAMPLES = """
```

Comment:
Can we include the module output in the examples?

Reply:
and also an example using vif?

Reply:
@cidrblock you mean module output of all the 4 operations? Something like:
```
# Using {{ operation }}
#
# Before state:
# --------------
#
#
# {{ before state config lines }}
#
#
#

{{ tasks example }}

# Module Run Output (i.e., contents of the `result` key)
# -------------------------------------------------------
#
#
# {{ module run output }}
# 
#
#

# After State
# ------------
#
#
# {{ after state config lines }}
#
#
```

> Can we include the module output in the examples?

Reply:
> and also an example using vif?

VIF example is added in merged operation example.